src/modules/user-settings/components/tests/editor-picker.test.tsx

```diff
@@ -0,0 +1,189 @@
+import { Reducer, UnknownAction } from '@reduxjs/toolkit';
+import { QueryStatus } from '@reduxjs/toolkit/query';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { produce } from 'immer';
+import { Provider } from 'react-redux';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+import { RootState, store } from 'src/stores';
+import { InstalledAppsState } from 'src/stores/installed-apps-api';
+import { testReducer } from 'src/stores/tests/utils/test-reducer';
+import { EditorPicker } from '../editor-picker';
+
+jest.mock( 'src/lib/get-ipc-api' );
+const mockGetIpcApi = getIpcApi as jest.Mock;
+
+// Create a test reducer for installedAppsApi
+function installedAppsTestReducer( state: RootState, action: UnknownAction ) {
+	if ( action.type === 'installedApps/setInstalledApps' ) {
+		const payload = action.payload as {
+			installedApps: InstalledAppsState;
+		};
+
+		return produce( state!, ( draftState ) => {
+			if ( draftState ) {
+				// Set the query result in the RTK Query cache
+				draftState.installedAppsApi.queries = {
+					'getInstalledApps({"forceRefetch":false})': {
+						status: QueryStatus.fulfilled,
+						data: payload.installedApps,
+						error: undefined,
+						requestId: 'test-request-id',
+						endpointName: 'getInstalledApps',
+						startedTimeStamp: 0,
+						fulfilledTimeStamp: 0,
+						originalArgs: undefined as never,
+					},
+				};
+			}
+		} );
+	}
+
+	return testReducer( state, action );
+}
+
+// Create test actions for installedAppsApi
+const installedAppsTestActions = {
+	setInstalledApps: ( installedApps: InstalledAppsState ) => {
+		return { type: 'installedApps/setInstalledApps', payload: { installedApps } };
+	},
+};
+
+// Replace the store's reducer with our test reducer
+store.replaceReducer( installedAppsTestReducer as Reducer< RootState > );
```

Comment:
```suggestion
store.replaceReducer( testReducer );
```
By using the `updateQueryData` helper, we can do away with the `installedAppsTestReducer`.

Reply:
I carried out these changes in https://github.com/Automattic/studio/pull/1259 and now it's merged to trunk, so these changes should be picked up after a rebase.